src/core/buffers/adaptation/adaptation_buffer.ts

```diff
@@ -163,8 +173,31 @@ export default function AdaptationBuffer<T>(
           observableOf(EVENTS.representationChange(adaptation.type,
                                                    period,
                                                    representation));
-        const representationBuffer$ = createRepresentationBuffer(representation)
-          .pipe(takeUntil(killCurrentBuffer$));
+
+        const bufferGoalRatio = bufferGoalRatioMap[representation.id] || 1;
+        bufferGoalRatioMap[representation.id] = bufferGoalRatio;
+
+        const bufferGoal$ = wantedBufferAhead$.pipe(
+          map((wba) => wba * bufferGoalRatio)
+        );
+
+        const representationBuffer$ = createRepresentationBuffer(representation,
+                                                                 bufferGoal$)
+        .pipe(
+          catchError((err) => {
+            if (err.code === "BUFFER_FULL_ERROR") {
+              const lastBufferGoalRatio = bufferGoalRatio;
+              if (lastBufferGoalRatio > 0.05) {
+                bufferGoalRatioMap[representation.id] = lastBufferGoalRatio - 0.05;
```

Comment:
Three things:
  1. Isn't that difference too low ? To arrive to `0.05` we have to apply that things 19 times !
  2. Maybe we should have a hard limit (like 2 seconds or something)
  3. This is only applied one time, ad the second time you call `createRepresentationBuffer`, it is not catched.